pkg/node/manager/manager.go

```diff
@@ -597,3 +597,13 @@ func (m *Manager) StartNeighborRefresh(nh datapath.NodeHandler) {
 	)
 	return
 }
+
+// CleanNeighbors cleans all neighbor entries for the direct routing device
+// and the encrypt interface. It should be used when the agent changes the
+// state from `n.enableNeighDiscovery = true` to
+// `n.enableNeighDiscovery = false`.
+func (m *Manager) CleanNeighbors(node datapath.NodeHandler) {
+	m.mutex.RWMutex.RLock()
+	defer m.mutex.RWMutex.RUnlock()
+	node.NodeCleanNeighbors()
```

Comment:
It's not entirely clear to me why this needs to be a separate method defined on the `nodeHandler`.  I don't see any references for the `receiver`.

Reply:
Good point, I've cleaned this up.